ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/errors/PaywallValidationError.kt

```diff
@@ -1,8 +1,35 @@
 package com.revenuecat.purchases.ui.revenuecatui.errors
 
+import com.revenuecat.purchases.Offering
+import com.revenuecat.purchases.ui.revenuecatui.strings.PaywallValidationErrorStrings
+
 sealed class PaywallValidationError : Throwable() {
-    object MissingPaywall : PaywallValidationError()
-    data class InvalidTemplate(val templateName: String) : PaywallValidationError()
-    data class InvalidVariables(val unrecognizedVariables: Set<String>) : PaywallValidationError()
-    data class InvalidIcons(val invalidIcons: Set<String>) : PaywallValidationError()
+
+    abstract fun associatedErrorString(offering: Offering): String
```

Comment:
Hmm instead of doing this an abstract function, you could implement it and use a `when(this) to return the appropriate string for each sealed class type.